### PR TITLE
Moved font-awesome challenges to Font Awesome v5.8.1

### LINKED ISSUE #35714: Move font-awesome challenges to Font Awesome 5.x.x
The challenges:

* [Add Font Awesome Icons to our Buttons](https://www.freecodecamp.rocks/learn/front-end-libraries/bootstrap/add-font-awesome-icons-to-our-buttons) , and,
* [Add Font Awesome Icons to all of our Buttons](https://www.freecodecamp.rocks/learn/front-end-libraries/bootstrap/add-font-awesome-icons-to-all-of-our-buttons)

Still teach Font Awesome v4.5.0. Although there's nothing wrong with teaching the older version, I personally feel that since it is just 2 challenges, we should update them to the latest version

### PR DESCRIPTION
**Pull request by wang1998tina**
- [ x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [ x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x] My pull request targets the `master` branch of freeCodeCamp.
- [x ] None of my changes are plagiarized from another source without proper attribution.
- [x ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [ x] My changes do not use shortened URLs or affiliate links.

Closes #35714

Since Font Awesome version 5.x.x introduces some new concepts (icon categories, etc), @dnntrn and I mostly just updated the existing content to support the new version.

### CODE REVIEW

**Review by ojeytonwilliams — APPROVED** [review]
If you have the time, it would be good if you could remove the unnecessary assert messages.  They are redundant and the string in `text` is used instead.  For example 
```
testString: assert($("i").is(".fas.fa-thumbs-up") || $("span").is(".fas.fa-thumbs-up"), 'Add an <code>i</code> element with the classes <code>fas</code> and <code>fa-thumbs-up</code>.');
```
 would become 
```
testString: assert($("i").is(".fas.fa-thumbs-up") || $("span").is(".fas.fa-thumbs-up"));
```
Similarly, if you could clean up the history by remove the merges and just leaving the commits were code is changed, that would be great.

The changes are improvements, nonetheless, so I'd be okay with it being merged as is.

File: curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-our-buttons.english.md [hunk 1]
```diff
@@ -2,21 +2,21 @@
...
+<code>&#60;link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous"&#62;</code>
 In this case, we've already added it for you to this page behind the scenes.
 The <code>i</code> element was originally used to make other elements italic, but is now commonly used for icons. You can add the Font Awesome classes to the <code>i</code> element to turn it into an icon, for example:
-<code>&lt;i class="fa fa-info-circle"&gt;&lt;/i&gt;</code>
+<code>&lt;i class="fas fa-info-circle"&gt;&lt;/i&gt;</code>
 Note that the <code>span</code> element is also acceptable for use with icons.
-Use Font Awesome to add a <code>thumbs-up</code> icon to your like button by giving it an <code>i</code> element with the classes <code>fa</code> and <code>fa-thumbs-up</code>; make sure to keep the text "Like" next to the icon.
+Use Font Awesome to add a <code>thumbs-up</code> icon to your like button by giving it an <code>i</code> element with the classes <code>fas</code> and <code>fa-thumbs-up</code>; make sure to keep the text "Like" next to the icon.
```
Comment (thecodingaviator) [inline curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-our-buttons.english.md]: Move this line to the instructions part

File: curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-all-of-our-buttons.english.md [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
+  - link: 'https://use.fontawesome.com/releases/v5.8.1/css/all.css'
     raw: true
 challengeType: 0
 ---
 
 ## Description
 <section id='description'>
-Font Awesome is a convenient library of icons. These icons are vector graphics, stored in the <code>.svg</code> file format. These icons are treated just like fonts. You can specify their size using pixels, and they will assume the font size of their parent HTML elements.
+Font Awesome is a convenient library of icons. These icons can be web fonts or vector graphics. These icons are treated just like fonts. You can specify their size using pixels, and they will assume the font size of their parent HTML elements.
 Use Font Awesome to add an <code>info-circle</code> icon to your info button and a <code>trash</code> icon to your delete button.
```
Comment (thecodingaviator) [inline curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-all-of-our-buttons.english.md]: Move lines 13 and 14 to the instructions section

**Review by thecodingaviator — DISMISSED** [review]
Check comments

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thanks for contributing to the freeCodeCamp repository. Congratulations on your first PR! 🎊 

Look forward to more contributions from you! ✈️

**Pull request merged**